addon/components/flexberry-boundingbox.js

```diff
@@ -0,0 +1,133 @@
+
+          this.changedByAreaSelect = true;
+          this.set('maxLgt', bounds.getEast());
+
+          this.changedByAreaSelect = true;
+          this.set('minLat', bounds.getSouth());
+
+          this.changedByAreaSelect = true;
+          this.set('maxLat', bounds.getNorth());
+          this.sendAction('boundingBoxChange', bounds);
+        });
+      }
+    })),
+
+  actions: {
+    /**
+    This action is called when latitude values are changed.
+
+    @method actions.latInputChange
+  */
+    latInputChange: function(event) {
+      if (this.changedByAreaSelect === true) {
+        this.changedByAreaSelect = false;
+        return;
+      }
+
+      if (event.target.value < -90) {
+        event.target.value = -90;
+      }
+
+      if (event.target.value > 90) {
+        event.target.value = 90;
+      }
+
+      let leafletMap = this.get('leafletMap');
+      let areaSelect = this.get('areaSelect');
+      let maxLat = Number(this.get('maxLat'));
+      let minLat = Number(this.get('minLat'));
+      let maxLgt = Number(this.get('maxLgt'));
+      let minLgt = Number(this.get('minLgt'));
+      let coords = L.latLng((maxLat + minLat) / 2, (maxLgt + minLgt) / 2);
+      leafletMap.panTo(coords);
+      leafletMap.fitBounds(L.latLngBounds(L.latLng(minLat, minLgt), L.latLng(maxLat, maxLgt)));
+      let newWidth = Math.abs(leafletMap.latLngToLayerPoint(L.latLng(minLat, maxLgt)).x) - Math.abs(leafletMap.latLngToLayerPoint(L.latLng(minLat, minLgt)).x);
+      let newHeight = Math.abs(leafletMap.latLngToLayerPoint(L.latLng(maxLat, minLgt)).y) - Math.abs(leafletMap.latLngToLayerPoint(L.latLng(minLat, minLgt)).y);
+      newWidth = Math.abs(newWidth);
+      newHeight = Math.abs(newHeight);
+      console.log('w: ', newWidth, 'h: ', newHeight);
+      areaSelect.setDimensions({ width: newWidth, height: newHeight });
+    },
+    /**
+    This action is called when longitude values are changed.
+
+    @method actions.lngInputChange
+  */
+    lngInputChange: function(event) {
+      if (this.changedByAreaSelect === true) {
+        this.changedByAreaSelect = false;
+        return;
+      }
+
+      if (event.target.value < -180) {
+        event.target.value = -180;
+      }
+
+      if (event.target.value > 180) {
+        event.target.value = 180;
+      }
+
+      let leafletMap = this.get('leafletMap');
+      let areaSelect = this.get('areaSelect');
+      let maxLat = Number(this.get('maxLat'));
+      let minLat = Number(this.get('minLat'));
+      let maxLgt = Number(this.get('maxLgt'));
+      let minLgt = Number(this.get('minLgt'));
+      let coords = L.latLng((maxLat + minLat) / 2, (maxLgt + minLgt) / 2);
+      leafletMap.panTo(coords);
+      leafletMap.fitBounds(L.latLngBounds(L.latLng(minLat, minLgt), L.latLng(maxLat, maxLgt)));
+      let newWidth = Math.abs(leafletMap.latLngToLayerPoint(L.latLng(minLat, maxLgt)).x) - Math.abs(leafletMap.latLngToLayerPoint(L.latLng(minLat, minLgt)).x);
```

Comment:
Am I stupid, or you are deducting the same numbers there and below?
Purpose?
besides, it looks really ugly.